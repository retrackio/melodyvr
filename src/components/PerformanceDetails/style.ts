import styled from 'styled-components';

export const Title = styled.p`
  font-family: 'Proza Libre';
  font-weight: 600;
  font-size: 1.8rem;
  color: #000;
  margin-bottom: 20px;
  margin-top: 50px;
`;

export const Description = styled.p`
  font-family: 'Open Sans';
  font-weight: 400;
  font-size: 1.6rem;
  color: #000;
  margin-bottom: 20px;
`;

export const Label = styled.p`
  font-family: 'Open Sans';
  font-weight: 400;
  font-size: 1.4rem;
  color: #000;
  margin-bottom: 20px;
`;

export const Debug = styled.pre`
  font-family: 'Consolas,monospace';
  font-size: 1.2em;
  overflow-x: auto;
  height: 250px;
  border: 1px solid #999;
  padding: 20px;
  margin: 20px 0px;
`;
