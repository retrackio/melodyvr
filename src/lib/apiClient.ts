import fetch from 'isomorphic-fetch';
import config from '../config';

export interface ReleasesResult {
  result: {
    releases: Release[];
  };
}

export interface Release {
  feature_order: number;
  is_free: boolean;
  product: Product;
}

export interface Product {
  id: number;
  title: string;
  description: string;
  image_hero_url: string;
  image_landscape_url: string;
  accent_colour_code: string;
  background_colour_code: string;
  text_colour_code: string;
  production_type_code: string;
  price: string | undefined;
}

export interface Venue {
  id: number;
  name: string;
  description: string;
  city: string;
  country_code: string;
}

export interface Artist {
  id: number;
  name: string;
  description: string;
  country_code: string;
  website_url: string;
  facebook_url: string;
  instagram_url: string;
  twitter_url: string;
  image_url: string;
  artist_background_colour_code: string;
  artist_text_colour_code: string;
  artist_accent_colour_code: string;
}

export interface Genre {
  id: number;
  name: string;
  description: string | null;
}

export type Performance = Product & {
  venue: Venue;
  artist: Artist;
  genre: Genre;
};

function sortReleases(releases: Release[]) {
  return releases.sort(
    (releaseA, releaseB) => releaseA.feature_order - releaseB.feature_order,
  );
}

export async function getReleases(): Promise<Release[]> {
  const response = await fetch(`${config.services.apiBaseUrl}/products`);
  const {
    result: { releases },
  } = await response.json();
  return sortReleases(releases);
}

export async function getPerformance(id: number): Promise<Performance> {
  const response = await fetch(`${config.services.apiBaseUrl}/products/${id}`);
  const { result } = await response.json();
  return result;
}
