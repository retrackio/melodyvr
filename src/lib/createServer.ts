import express from 'express';
import Server from 'next/dist/next-server/server/next-server';

export default function createServer(app: Server) {
  const server = express();
  const handle = app.getRequestHandler();

  server.all(
    '*',
    (req: express.Request, res: express.Response) => handle(req, res),
  );

  return server;
}
