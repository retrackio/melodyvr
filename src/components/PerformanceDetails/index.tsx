import React from 'react';
import { Performance } from '../../lib/apiClient';
import * as Styled from './style';

interface Props {
  performance: Performance;
}

export default function PerformanceDetails({ performance }: Props) {
  return (
    <>
      <Styled.Title>{performance.title}</Styled.Title>
      <Styled.Description
        dangerouslySetInnerHTML={{ __html: performance.description }}
      />
      <Styled.Label>
        Facebook:{' '}
        <a
          href={performance.artist.facebook_url}
          target="_blank"
          rel="noreferrer"
        >
          {performance.artist.facebook_url}
        </a>
      </Styled.Label>
      <Styled.Label>
        Instagram:{' '}
        <a
          href={performance.artist.instagram_url}
          target="_blank"
          rel="noreferrer"
        >
          {performance.artist.instagram_url}
        </a>
      </Styled.Label>
      <Styled.Label>Raw data:</Styled.Label>
      <Styled.Debug>{JSON.stringify(performance, null, 4)}</Styled.Debug>
    </>
  );
}
