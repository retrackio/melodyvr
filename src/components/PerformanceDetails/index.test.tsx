import React from 'react';
import { shallow } from 'enzyme';
import PerformanceDetails from './';
import { Performance } from '../../lib/apiClient';

const mockedPerformance: Performance = {
  id: 2,
  title: 'Imagine Dragons',
  description:
    '<i>Performance contains flashing or strobe lighting.</i>\n\nThe biggest rock band of 2018 perform their eagerly anticipated new album from Red Rocks Amphitheatre, Colorado. Expect ground-shaking tracks, huge vocals and even bigger confetti cannonsâ€¦',
  image_hero_url:
    'https://d17jafawxl91z1.cloudfront.net/IMD_P0129_IMG_Hero.jpg',
  image_landscape_url:
    'https://d17jafawxl91z1.cloudfront.net/IMD_P0129_IMG_Landscape.jpg',
  accent_colour_code: 'F52459',
  background_colour_code: '1e1e1e',
  text_colour_code: 'ffffff',
  production_type_code: 'performance',
  price: '9.99',
  venue: {
    id: 1508,
    name: 'Red Rocks Amphitheatre',
    description: '',
    city: 'Colorado',
    country_code: 'US',
  },
  artist: {
    id: 2145,
    name: 'Imagine Dragons',
    description:
      'Imagine Dragons is an American rock band from Las Vegas, Nevada, consisting of lead vocalist Dan Reynolds, lead guitarist Wayne Sermon, bassist and keyboardist Ben McKee, and drummer Daniel Platzman.',
    country_code: 'US',
    website_url: 'https://www.imaginedragonsmusic.com/',
    facebook_url: 'https://www.facebook.com/ImagineDragons/',
    instagram_url: 'https://www.instagram.com/imaginedragons/?hl=en',
    twitter_url: 'https://twitter.com/Imaginedragons',
    image_url:
      'https://d17jafawxl91z1.cloudfront.net/58e8398bb1229b89fc9fa1c55f77e6bf_IMD_P0063_IMG_Square.jpg',
    artist_background_colour_code: '1E1E1E',
    artist_text_colour_code: 'FFFFFF',
    artist_accent_colour_code: 'BA8F65',
  },
  genre: {
    id: 680,
    name: 'Alternative',
    description: null,
  },
};

describe('<PerformanceDetails />', () => {
  it('renders correctly', () => {
    const wrapper = shallow(
      <PerformanceDetails performance={mockedPerformance} />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
