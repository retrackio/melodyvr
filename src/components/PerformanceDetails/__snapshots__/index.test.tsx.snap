// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PerformanceDetails /> renders correctly 1`] = `
<Fragment>
  <style__Title>
    Imagine Dragons
  </style__Title>
  <style__Description
    dangerouslySetInnerHTML={
      Object {
        "__html": "<i>Performance contains flashing or strobe lighting.</i>

The biggest rock band of 2018 perform their eagerly anticipated new album from Red Rocks Amphitheatre, Colorado. Expect ground-shaking tracks, huge vocals and even bigger confetti cannons…",
      }
    }
  />
  <style__Label>
    Facebook:
     
    <a
      href="https://www.facebook.com/ImagineDragons/"
      rel="noreferrer"
      target="_blank"
    >
      https://www.facebook.com/ImagineDragons/
    </a>
  </style__Label>
  <style__Label>
    Instagram:
     
    <a
      href="https://www.instagram.com/imaginedragons/?hl=en"
      rel="noreferrer"
      target="_blank"
    >
      https://www.instagram.com/imaginedragons/?hl=en
    </a>
  </style__Label>
  <style__Label>
    Raw data:
  </style__Label>
  <style__Debug>
    {
    "id": 2,
    "title": "Imagine Dragons",
    "description": "&lt;i&gt;Performance contains flashing or strobe lighting.&lt;/i&gt;\\n\\nThe biggest rock band of 2018 perform their eagerly anticipated new album from Red Rocks Amphitheatre, Colorado. Expect ground-shaking tracks, huge vocals and even bigger confetti cannons…",
    "image_hero_url": "https://d17jafawxl91z1.cloudfront.net/IMD_P0129_IMG_Hero.jpg",
    "image_landscape_url": "https://d17jafawxl91z1.cloudfront.net/IMD_P0129_IMG_Landscape.jpg",
    "accent_colour_code": "F52459",
    "background_colour_code": "1e1e1e",
    "text_colour_code": "ffffff",
    "production_type_code": "performance",
    "price": "9.99",
    "venue": {
        "id": 1508,
        "name": "Red Rocks Amphitheatre",
        "description": "",
        "city": "Colorado",
        "country_code": "US"
    },
    "artist": {
        "id": 2145,
        "name": "Imagine Dragons",
        "description": "Imagine Dragons is an American rock band from Las Vegas, Nevada, consisting of lead vocalist Dan Reynolds, lead guitarist Wayne Sermon, bassist and keyboardist Ben McKee, and drummer Daniel Platzman.",
        "country_code": "US",
        "website_url": "https://www.imaginedragonsmusic.com/",
        "facebook_url": "https://www.facebook.com/ImagineDragons/",
        "instagram_url": "https://www.instagram.com/imaginedragons/?hl=en",
        "twitter_url": "https://twitter.com/Imaginedragons",
        "image_url": "https://d17jafawxl91z1.cloudfront.net/58e8398bb1229b89fc9fa1c55f77e6bf_IMD_P0063_IMG_Square.jpg",
        "artist_background_colour_code": "1E1E1E",
        "artist_text_colour_code": "FFFFFF",
        "artist_accent_colour_code": "BA8F65"
    },
    "genre": {
        "id": 680,
        "name": "Alternative",
        "description": null
    }
}
  </style__Debug>
</Fragment>
`;
